### PR TITLE
MPD.new(): Read standard environmental variables

### PR DESCRIPTION
**Pull request by doronbehar**
Parse $MPD_HOST when evaluating password and host.

### CODE REVIEW

File: mpd.lua [hunk 1]
```diff
@@ -26,13 +26,17 @@ local MPD = {}
...
+    local env_host = os.getenv('MPD_HOST') or 'localhost'
+    local _, _, env_password, env_hostname = string.find(env_host, "([^@]+)@([a-zA-Z0-9.]+)")
+
     local client = {}
     if settings == nil then settings = {} end
 
-    client.hostname = settings.hostname or "localhost"
-    client.port     = settings.port or 6600
+    client.hostname = settings.hostname or env_hostname
+    client.port     = settings.port or os.getenv('MPD_PORT') or 6600
```
Comment (kaworu) [inline mpd.lua]: I think we need `tonumber()` the `MPD_PORT` env variable

**Review by kaworu — CHANGES_REQUESTED** [review]

**Comment by kaworu** [comment]
Thanks for this PR, I like the idea! See my concern about the port settings, otherwise LGTM.

**Comment by doronbehar** [comment]
Agreed.

**Comment by kaworu** [comment]
@doronbehar Thanks! I'll try to find the time to test, merge and release this weekend :)

**Review by kaworu — APPROVED** [review]

**Comment by kaworu** [comment]
@doronbehar Hi there, sorry for the delay.

So after a quick test, this patch fail when `MPD_HOST` is set to `localhost` (or not set since it is the default). The regexp find fail, and then `env_hostname` is `nil` which cause connect to fail.

Since it is becoming heavier, I think it would be cleaner to extract the env parsing stuff to a function, so that we can do something like that from the constructor:
```lua
env_hostname, env_port, env_password = mpd_getenv()
```

Also, I would love to see the constructor comment and README updated accordingly.

Best Regards,
Alex

File: mpd.lua [hunk 2]
```diff
@@ -26,13 +26,17 @@ local MPD = {}
 --      timeout:  time in sec to wait for connect() and receive() (default 1)
 --      retry:    time in sec to wait before reconnect if error (default 60)
 function MPD.new(settings)
+
+    local env_host = os.getenv('MPD_HOST') or 'localhost'
+    local _, _, env_password, env_hostname = string.find(env_host, "([^@]+)@([a-zA-Z0-9.]+)")
```
Comment (kaworu) [inline mpd.lua]: After this, `env_hostname` could be `nil`

**Review by kaworu — CHANGES_REQUESTED** [review]

**Comment by doronbehar** [comment]
Totally agree with you, I've pushed several commits which hopefully fix it.

One question: As for parsing the hostname part of `$MPD_HOST`, what do you think about using a regular expression such as this: https://stackoverflow.com/a/106223/4935114 ?

**Comment by kaworu** [comment]
Hi @doronbehar,

So I've tested and the patch is working fine now when `$MPD_HOST` and/or `$MPD_PORT` are not defined.

Not a big fan of the regexp from SO as `MPD_HOST` may be a hostname, IPv4, IPv6, or even a path to a Unix domain socket. Note that at the moment we don't support IPv6 (because lua-socket doesn't have released support for it AFAIK) nor Unix domain socket (could be done if someone wanted to). So the regexp from your patch looks fine to me.

Thanks !

**Comment by kaworu** [comment]
@doronbehar https://luarocks.org/modules/kAworu/mpd/1.2.0-1

**Pull request merged**